Tests/Shared.Specs/MethodInfoAssertionSpecs.cs

```diff
@@ -125,6 +126,49 @@ public void When_asserting_a_method_is_decorated_with_attribute_and_it_is_it_suc
             act.Should().NotThrow();
         }
 
+        [Fact]
+        public void When_asserting_a_method_is_decorated_with_MethodImpl_attribute_and_it_is_it_succeeds()
+        {
+            //-------------------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-------------------------------------------------------------------------------------------------------------------
+            MethodInfo methodInfo = typeof(ClassWithMethodWithImplementationAttribute).GetParameterlessMethod("DoNotInlineMe");
+
+            //-------------------------------------------------------------------------------------------------------------------
+            // Act
+            //-------------------------------------------------------------------------------------------------------------------
+            Action act = () =>
+                methodInfo.Should().BeDecoratedWith<MethodImplAttribute>();
+
+            //-------------------------------------------------------------------------------------------------------------------
+            // Assert
+            //-------------------------------------------------------------------------------------------------------------------
+            act.Should().NotThrow();
+        }
+
+        [Fact]
+        public void When_asserting_a_method_is_decorated_with_MethodImpl_attribute_and_it_is_not_it_throws()
+        {
+            //-------------------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-------------------------------------------------------------------------------------------------------------------
+            MethodInfo methodInfo = typeof(ClassWithAllMethodsDecoratedWithDummyAttribute).GetParameterlessMethod("PublicDoNothing");
+
+            //-------------------------------------------------------------------------------------------------------------------
+            // Act
+            //-------------------------------------------------------------------------------------------------------------------
+            Action act = () =>
+                methodInfo.Should().BeDecoratedWith<MethodImplAttribute>();
+
+            //-------------------------------------------------------------------------------------------------------------------
+            // Assert
+            //-------------------------------------------------------------------------------------------------------------------
+            act.Should().Throw<XunitException>()
```

Comment:
By replacing `WithMessage(...)` with something like `WithMessage("*PublicDoNothing*MethodImplAttribute*")` the test becomes more robust.
E.g if someone wanted to reword the failure message to end with ", but could not find that attribute.".

Reply:
By "everywhere", do you mean everywhere in Fluent Assertions?
A quick look yesterday showed that at least 30% of `WithMessage` uses wildcards. 

By including the entire message in the assertion, one essentially asserts that the exception under test must have a very specific message. 

I'll put cleaning up exception message assertions on my todo. It is not very fair to contributors that the existing code is written in one way, but we tell them to write it another way.

Reply:
I meant everywhere in the parts I was changing. Turns out it's not a very good sample for the population then.